### PR TITLE
Major reengineering to make demos collected from different sources compatible 

### PR DESCRIPTION
**Pull request by ankile**
Shoutout to @anthonysimeonov!

The main change is to make changes around the OSC controller to make stored actions, regardless of source, be the actions that are actually executed—not exaggerated actions that are internally scaled in the controller as before.

This change makes the demos from the scripted demos, teleop, and rollouts play nicer together, a prerequisite for this project to work.

At the same time, we seized the opportunity to do a larger reengineering of some legacy systems based on learning from the first build, including:
- Treat image observations differently: simulate in full res and resize down
- Store images slightly bigger than needed to have better crop augmentation
- Make a dataset system that is modularized so one can mix and match data easily without having a lot of duplication and need to recompute precomputed features over and over
- Improve the teleop tool with, e.g., an undo button and the opportunity to continue trajectories
- Start a trajectory augmentation tool for creating better coverage and a funnel around bottleneck states
- Script for resuming runs easily
- Switch to using 6D rotation representation
- Enable easier experimentation with position control

And there's more to come!

### CODE REVIEW

**Comment by ankile** [comment]
I added you as a reviewer in case you're finding yourself with too much free time and want to have a look, @anthonysimeonov, but absolutely no pressure to actually review!

File: src/data_collection/teleop.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    data_path = trajectory_save_dir(
+        environment="sim" if args.is_sim else "real",
+        task=args.furniture,
+        demo_source="teleop",
+        randomness=args.randomness,
+    )
+
+    from pathlib import Path
+
+    pickle_paths = list(
```
Comment (anthonysimeonov) [inline src/data_collection/teleop.py:86]: Want to make this kind of demo loading optional + not have the path here hard coded?

File: src/train/train_bc.py [hunk 2]
```diff
@@ -368,21 +367,23 @@ def get_data_path(obs_type, encoder, task, suffix=None):
...
-    config.furniture = args.furniture
+
+    # TODO: Change this to "task"
+    # config.furniture = args.furniture
+    # config.n_parts_assemble = args.n_parts_assemble
+
     config.gpu_id = args.gpu_id
     config.load_checkpoint_path = None
-    config.load_checkpoint_path = "/data/scratch/ankile/furniture-diffusion/models/trim-wave-5/actor_chkpt_best_test_loss.pt"
+    # config.load_checkpoint_path = "/data/scratch/ankile/furniture-diffusion/models/ruby-butterfly-17/actor_chkpt_best_test_loss.pt"
```
Comment (anthonysimeonov) [inline src/train/train_bc.py:386]: When loading a checkpoint, do we also need to ensure we manually enter the correct `config.wandb.continue_run_id` value above?

**Review by anthonysimeonov — COMMENTED** [review]
left a couple of minor comments, but overall looks great!

File: src/data_collection/teleop.py (hunk 1 above)
Comment (ankile) [reply src/data_collection/teleop.py:86]: Yes, absolutely, will figure out the best way to make this as convenient as possible!

File: src/train/train_bc.py (hunk 2 above)
Comment (ankile) [reply src/train/train_bc.py:386]: I'm revamping the run launching system as it feels necessary is there are more and more different experiments going on, so think I might just embrace Hydra at this point.

**Pull request merged**